### DISCUSSION TITLE
Load the session from the main branch if no branch session

### DISCUSSION THREAD

**Discussion by akinsho** [olimorris/persisted.nvim#65] [Q&A] [answered]
So I'm using persisted with the setting to save session per branch which often means that when I open neovim if I've created the branch outside of neovim there won't be an associated session and neovim will open into an empty buffer. I personally often want it to instead start with the main branches session since that's most likely to exist so I can start from there rather than a blank session.

when using `SessionLoadLast` inside of the `no_session` callback though this will sometimes load a session from a completely different project. I'm not sure if I can use the `SessionFromFile` command though since I want it to load from the `main` branch not necessarily the directory.

Any ideas?

**Comment by olimorris** [comment]
Yeah this is a really good point. I have come across this before myself, but new branches are a bit of a rare one now.

We could detect if a session exists for the `main` branch (or possibly the last save session for the repo).

Could also introduced a new callback but likely to be complex for users to utilise.

**Reply by akinsho** [reply]
@olimorris I think if there was an option to detect if there is a `main` (then falling back to `master`) would be awesome. I think it could be as easy as adding an option although if you are worried about option "bloat", it could also maybe be that there is some convenience mechanism for getting file to use thought that sounds a bit longwinded

**Reply by olimorris** [reply, marked as answer]
Can you try out #66 and feedback on the PR?
